collab-folder/src/core/view.rs

```diff
@@ -478,8 +503,51 @@ pub struct View {
   #[serde(default)]
   pub is_favorite: bool,
   pub layout: ViewLayout,
-  pub icon_url: Option<String>,
-  pub cover_url: Option<String>,
+  pub icon: Option<ViewIcon>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+pub enum IconType {
```

Comment:
Can we use Serialize_repr and Deserialize_repr? Please refer to the definition of DatabaseLayout